src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.iOS.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace System.IO
+{
+    public static partial class Path
+    {
+        private static string? s_defaultTempPath;
+
+        private static string? DefaultTempPath
```

Comment:
why does this need to be nullable? can we make it non-nullable instead?

Reply:
Presumably `CS8618: Non-nullable field 's_defaultTempPath' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.`

`SystemNative_SearchPath_TempDirectory` can explicitly return `NULL`, so `Interop.Sys.SearchPathTempDirectory` returns `string?`, so `DefaultTempPath` returns `string?`

Reply:
That might be true but the only place where it's used you're doing `DefaultTempPath!` i.e. essentially ignoring the NULL value and that would make `GetTempPath()` return NULL too which is unexpected.

I'd rather throw an exception when `Interop.Sys.SearchPathTempDirectory()` returns null (which should never happen) and make this non-nullable.
Regarding the CS8618 you can assign an empty string to initialize it.

Reply:
Which exception types which are safe to use inside corlib would be appropriate here? ExternalException?